### ISSUE TITLE
`laya-core.js` skips `clamp_temperature`, inflating confidence for 11+ option choices

### ISSUE THREAD

**Issue by jgrunik** [vishalmysore/layaForWeb#1]
## What the defect is

`web/laya-core.js` scales the logits by the raw temperature table value:

```js
const tScale = cfg.temperature_by_options?.[tempBucket(qt, k)] ?? cfg.temperature[qt];
```

Python's `laya/common.py` runs the same value through `clamp_temperature()`, which confines it to
`[0.5, 5.0]` (falling back to `1.0` for non-numeric, `NaN`, or `±inf` input). The browser port
omits that clamp.

The shipped checkpoint's `temperature_by_options` contains
`"choice:11+": 0.10058280825614929` — the only bucket outside `[0.5, 5.0]`. Dividing by it
raw sharpens the logits roughly 5× and inflates confidence for any choice question with 11+
options. Python emits a warning about this bucket; the port silently uses it.

## Why it matters

The repo's own demo gates automation on confidence (default threshold 0.90). An inflated
confidence can carry a 12-option decision over a gate it should not clear, and any downstream
port or copy of this file inherits the divergence.

## Status (verified)

- Deployed demo `https://vishalmysore.github.io/layaForWeb/laya-core.js` — **affected**; it
  serves the unclamped path.
- Upstream `main` HEAD `fde17d11175f9073c964c642dc02b4ffd11beaa9` — **affected**; no clamp exists.
- No prior issue, PR, or commit addresses `clamp_temperature` in this repository.

## Minimal reproduction

One 12-option choice question (`choice:11+`), q8e8 int8 ONNX build, run under
`onnxruntime-node`:

```
raw temperature        0.10058280825614929   (used unclamped)
unclamped confidence   0.998
clamped confidence     0.654                 (delta 0.344)
top label              unchanged (index 2, "application")
```

q8e8 is **int8-quantized**, so the meaningful check is the top label and a comparable confidence
*magnitude*, not exact equality with the fp32 reference. For reference, the PyTorch build for the
same input gives confidence 0.5513 with the same top label; the clamped result (0.654) is in the
same range, the unclamped one (0.998) is not.

## Patch

Unified diff against upstream `main` HEAD `fde17d1`. Confirmed applies cleanly with
`git apply --check` against that commit; applying it reproduces the fixed file byte-for-byte.

```diff
--- a/web/laya-core.js
+++ b/web/laya-core.js
@@ -74,6 +74,23 @@
   const size = k <= 2 ? "2" : k <= 5 ? "3-5" : k <= 10 ? "6-10" : "11+";
   return `${QTYPE_NAMES[qt]}:${size}`;
 }
+
+// Exact port of Python's clamp_temperature (laya/common.py:232). The shipped checkpoint has
+// choice:11+ = 0.10058, the one bucket outside [0.5, 5]; using it raw sharpens those logits ~5x
+// and inflates confidence for choice questions with 11+ options (Python warns about this bucket).
+//
+// Mirrors Python's float() coercion: non-numeric input (None, "", unparseable) falls back to 1.0;
+// NaN/±inf also fall back to 1.0; everything else is clamped into [TEMP_MIN, TEMP_MAX].
+const TEMP_MIN = 0.5, TEMP_MAX = 5.0;
+export function clampTemperature(t) {
+  let v;
+  if (typeof t === "number") v = t;
+  else if (typeof t === "string" && t.trim() !== "") v = Number(t);
+  else return 1.0;
+  if (!Number.isFinite(v)) return 1.0;
+  return Math.min(TEMP_MAX, Math.max(TEMP_MIN, v));
+}
+
 export function confidenceFromProbs(p, k) {
   if (k < 2) return 1.0;
   let ent = 0;
@@ -131,8 +148,8 @@
     const answers = {};
     ids.forEach((qid, r) => {
       const q = qs[r], k = items[r].markers.length, qt = items[r].qtype;
-      const tScale = cfg.temperature_by_options?.[tempBucket(qt, k)] ?? cfg.temperature[qt];
-      const z = Array.from({ length: k }, (_, i) => logits[r * b.kmax + i] / Math.max(1e-3, tScale));
+      const tScale = clampTemperature(cfg.temperature_by_options?.[tempBucket(qt, k)] ?? cfg.temperature[qt]);
+      const z = Array.from({ length: k }, (_, i) => logits[r * b.kmax + i] / tScale);
       const zmax = Math.max(...z);
       const e = z.map((v) => Math.exp(v - zmax));
       const s = e.reduce((a, v) => a + v, 0);
```

## Notes

- The same `temperature_by_options` table (including `choice:11+`) ships in the Hugging Face model
  `convaiinnovations/laya-typed-decisions`, so switching checkpoints does not avoid it.
- The repo's own 108-case suite contains only 3-, 4-, and 5-option choice questions (24, 60, and
  12 respectively), so `choice:11+` is never exercised — very likely why this went unnoticed.

**Comment by vishalmysore** [comment]
Thanks for the clear report and the patch. Confirmed: both `convaiinnovations/laya` and `laya-typed-decisions` ship `choice:11+ = 0.1006`, and upstream `laya` 0.3.7 clamps it in `clamp_temperature`, which the browser port skipped.

Fixed in 6e67de8:
- `web/laya-core.js`: added `clampTemperature()` (your patch, mirroring Python's `[0.5, 5.0]` range with a 1.0 fallback for non-finite or non-numeric values) and applied it before the softmax.
- `scripts/verify_model.py`: now applies the same `clamp_temperature`, so the parity numbers match real inference.

Only choice questions with 11+ options were affected. Other buckets are already in range and are unchanged. Closing, but feel free to reopen if you see anything off.
